tests/unit/package_managers/test_gomod.py

```diff
@@ -665,8 +798,107 @@ def test_parse_workspace_modules(
     assert parsed_workspace == expected_module
 
 
+@pytest.mark.parametrize(
+    "go_work_edit_json, relative_file_paths",
+    [
+        (
+            # main module is the same as the source dir, there's one nested workspace
+            """
+            {
+                "Use": [
+                    {"DiskPath": "."},
+                    {"DiskPath": "./workspace"}
+                ]
+            }
+            """,
+            ["./go.sum", "./workspace/go.sum", "./go.work.sum"],
+        ),
+        (
+            # go.work is in the source dir, main module and a workspace are nested
+            """
+            {
+                "Use": [
+                    {"DiskPath": "./app"},
+                    {"DiskPath": "./workspace"}
+                ]
+            }
+            """,
+            ["./app/go.sum", "./workspace/go.sum", "./go.work.sum"],
+        ),
+    ],
+)
+@mock.patch("subprocess.run")
+def test_get_go_sum_files(
+    mock_run: mock.Mock, go_work_edit_json: str, relative_file_paths: list[str], tmp_path: Path
+) -> None:
+    mock_run.side_effect = [
+        proc_mock("go work -edit -json", returncode=0, stdout=go_work_edit_json)
+    ]
+    go_work_root = RootedPath(tmp_path)
+
+    files = _get_go_sum_files(go_work_root, Go(), {})
+
+    expected_files = [go_work_root.join_within_root(path) for path in relative_file_paths]
+    assert files == expected_files
+
+
+def test_get_go_work_root(tmp_path: Path) -> None:
+    go_work_file = dedent(
+        """
+        go 1.19
+
+        use (
+            ./app
+            ./workspace
+        )
+        """
+    )
+    tmp_path.joinpath("go.work").write_text(go_work_file)
+
+    repo_root = RootedPath(tmp_path)
+
+    go_work_root = _get_go_work_root(repo_root, Go(), {"cwd": repo_root})
```

Comment:
Is this calling go directly? Should we be mocking that?

Reply:
I lost track of where exactly this change was, but I believe we have changed it to use a mocked `Go._run`.